lib/utils/normalizeOptions.js

```diff
@@ -104,6 +104,18 @@ function normalizeOptions(compiler, options, logger) {
     options.client = {};
   }
 
+  if (typeof options.client.webSocketURL === 'undefined') {
+    options.client.webSocketURL = {};
+  } else if (typeof options.client.webSocketURL === 'string') {
+    const parsedURL = new URL(options.client.webSocketURL);
+
+    options.client.webSocketURL = {
+      host: parsedURL.hostname,
+      port: parsedURL.port,
```

Comment:
Hmm, I think port here will be type `string`,  schema have only type `number`

Reply:
Yes, I think about it, here two solution:
- allow to set port as `string` everywhere
- do not allow use `port` as `string` everywhere

Let's fix it in other PR, I found we allow to set `host` to `null`, it is wrong, also we need allow to set port as `string` for websocket server too

Reply:
> allow to set port as string everywhere

+1, I can send a PR for the same 👍🏻